analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs

```diff
@@ -59,28 +58,76 @@ protected RequestsWithExcessiveLengthBase(System.Resources.ResourceManager rspec
+                            {
+                                cc.ReportDiagnosticWhenActive(Diagnostic.Create(rule, attribute.GetLocation()));
+                            }
+
+                            if ((IsInvalidRequestSizeLimit(attribute, cc.SemanticModel)
+                                 || IsInvalidRequestFormLimits(attribute, cc.SemanticModel))
+                                && GetMethodOrClassDeclaration(attribute, cc.SemanticModel) is { } delcaration)
+                            {
+                                var isRequestFormLimits = attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_RequestFormLimitsAttribute, cc.SemanticModel);
+
+                                attributesOverTheLimit.AddOrUpdate(
+                                    delcaration,
+                                    new Attributes
+                                    {
+                                        InvalidRequestForm = isRequestFormLimits ? attribute : null,
+                                        InvalidRequestSize = isRequestFormLimits ? null : attribute
+                                    },
+                                    (declaration, attributes) =>
+                                    new Attributes
+                                    {
+                                        InvalidRequestForm = isRequestFormLimits ? attribute : attributes.InvalidRequestForm,
+                                        InvalidRequestSize = isRequestFormLimits ? attributes.InvalidRequestSize : attribute
+                                    });
+                            }
+                        },
+                        Language.SyntaxKind.Attribute);
+
+                    c.RegisterCompilationEndAction(
+                        cc =>
+                        {
+                            foreach (var attrsByDec in attributesOverTheLimit)
+                            {
+                                var mainAttribute = attrsByDec.Value.InvalidRequestForm ?? attrsByDec.Value.InvalidRequestSize;
+
+                                cc.ReportDiagnosticWhenActive(
+                                    attrsByDec.Value.InvalidRequestForm != null
+                                    && attrsByDec.Value.InvalidRequestSize != null
+                                        ? Diagnostic.Create(rule, mainAttribute.GetLocation(), new List<Location> { attrsByDec.Value.InvalidRequestSize.GetLocation() })
+                                        : Diagnostic.Create(rule, mainAttribute.GetLocation()));
+                            }
+                        });
+                });
 
         private bool IsEnabled(AnalyzerOptions options)
         {
             analyzerConfiguration.Initialize(options);
             return SupportedDiagnostics.Any(d => analyzerConfiguration.IsEnabled(d.Id));
         }
+
+        private struct Attributes
+        {
+            public SyntaxNode InvalidRequestForm { get; set; }
```

Comment:
do the setters need to be public? can't you just make a constructor for the initialization,  and then have getter-only properties?

from what I see, you don't overwrite the values, so this should be immutable